newIDE/app/src/ObjectEditor/Editors/Model3DEditor.js

```diff
@@ -260,32 +286,136 @@ const Model3DEditor = ({
     {}
   );
 
-  const [model3D, setModel3D] = React.useState<?GLTF>(null);
-  const getModel3D = React.useCallback(
+  const onChangeProperty = React.useCallback(
+    (property: string, value: string) => {
+      objectConfiguration.updateProperty(property, value);
+      forceUpdate();
+    },
+    [objectConfiguration, forceUpdate]
+  );
+
+  // This doesn't loop indefinitely because the loader always return the same
+  // instance of GLTF.
+  const [gltf, setGltf] = React.useState<GLTF | null>(null);
+  const getGltf = React.useCallback(
     (modelResourceName: string) => {
       PixiResourcesLoader.get3DModel(project, modelResourceName).then(
         newModel3d => {
-          setModel3D(newModel3d);
+          setGltf(newModel3d);
         }
       );
     },
     [project]
   );
-  getModel3D(properties.get('modelResourceName').getValue());
+  getGltf(properties.get('modelResourceName').getValue());
+  const onChangeModelResourceName = React.useCallback(
+    (modelResourceName: string) => {
+      getGltf(modelResourceName);
+    },
+    [getGltf]
+  );
 
-  const onChangeProperty = React.useCallback(
-    (property: string, value: string) => {
-      objectConfiguration.updateProperty(property, value);
-      forceUpdate();
+  const model3D = React.useMemo<THREE.Object3D | null>(
```

Comment:
Is there a risk of memory leak? Is there a destroy/delete operation to do with three.js?

Reply:
I guess it should be fine, I don't see any method to destroy an Object3D in the doc. I guess an object is just a plain model (here, it's used to evaluate the box, it's not even used with a renderer)
https://threejs.org/docs/#api/en/core/Object3D